packager/signer/validation.go

```diff
@@ -123,10 +123,34 @@ func fetchURLMatches(url *url.URL, pattern *util.URLPattern) error {
 	return urlMatches(url, *pattern)
 }
 
+// Determines if b is either a valid byte of a fallback URL, per
+// https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#seccons-content-sniffing
+// item 3.1, or is the U+0025 (%) character.
+func isFallbackURLCodePoint(b byte) bool {
+	// https://url.spec.whatwg.org/#url-code-points, but in codepoint order:
+	//
+	// U+0021 (!), U+0024 ($), U+0026 (&), U+0027 ('), U+0028 LEFT PARENTHESIS,
+	// U+0029 RIGHT PARENTHESIS, U+002A (*), U+002B (+), U+002C (,), U+002D (-),
+	// U+002E (.), U+002F (/), ASCII digits (U+0030 - U+0039), U+003A (:),
+	// U+003B (;), U+003D (=), U+003F (?), U+0040 (@),
+	// ASCII upper alpha (U+0041 - U+005A), U+005F (_),
+	// ASCII lower alpha (U+0061 - U+007A), U+007E (~)
+
+	// Vaguely ordered most to least common, to aid short-circuiting:
+	return (b >= 'a' && b <= 'z') || b == '_' || b == '~' ||
+		(b >= '!' && b < 'Z' && b != '"' /*x22*/ && b != '#' /*x23*/ && b != '<' /*x3C*/ && b != '>' /*x3E*/)
+}
+
 // True iff url matches pattern, as defined by an [URLSet.Sign] block in the
 // config file. The format of this URLPattern is validated by
 // validateSignURLPattern in config.go.
 func signURLMatches(url *url.URL, pattern *util.URLPattern) error {
+	for _, b := range []byte(url.String()) {
```

Comment:
Optional, leverage Go's IndexFunc, e.g. if IndexFun(url.String(), isFallbackURLCodePoint) == -1 { ... }

Reply:
Could work? https://golang.org/pkg/bytes/#IndexFunc is based on a UTF-8 decoding of the string, whereas I want to go byte-by-byte. I think any multibyte UTF-8 char is also false for `isFallbackURLCodePoint`? But I don't want to have to think about it. :)